scripts/travis/upload-to-quay.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set -e
+
+BRANCH=${BRANCH:?'missing BRANCH env var'}
+IMAGE="${REPO:?'missing REPO env var'}:latest"
+
+unset major minor patch
+if [[ "$BRANCH" == "master" ]]; then
+  TAG="latest"
+elif [[ $BRANCH =~ ^v([0-9]+)\.([0-9]+)\.([0-9]+)$ ]]; then
+  major="${BASH_REMATCH[1]}"
+  minor="${BASH_REMATCH[2]}"
+  patch="${BASH_REMATCH[3]}"
+  TAG=${major}.${minor}.${patch}
+  echo "BRANCH is a release tag: major=$major, minor=$minor, patch=$patch"
+else
+  # TODO why do we do /// ?
+  TAG="${BRANCH///}"
+fi
+echo "TRAVIS_BRANCH=$TRAVIS_BRANCH, REPO=$REPO, BRANCH=$BRANCH, TAG=$TAG, IMAGE=$IMAGE"
+
+# add major, major.minor and major.minor.patch tags
+if [[ -n $major ]]; then
+  docker tag $IMAGE $REPO:${major}
+  if [[ -n $minor ]]; then
+    docker tag $IMAGE $REPO:${major}.${minor}
+    if [[ -n $patch ]]; then
+        docker tag $IMAGE $REPO:${major}.${minor}.${patch}
+    fi
+  fi
+fi
+
+if [[ -f $HOME/.docker/config.json ]]; then
+  rm -f $HOME/.docker/config.json
+else
+  echo "$HOME/.docker/config.json doesn't exist"
+fi
+
+# Do not enable echo before the `docker login` command to avoid revealing the password.
+set -x
+docker login quay.io -u $QUAY_USER -p $QUAY_PASS 
+echo "Quay login successful"
+
+# Remove org name jaegertracing to get image name
+function strip_image_name {
+  SUBSTRING_1=$(echo $1| cut -d'/' -f 2)
+  SUBSTRING_2=$(echo $SUBSTRING_1| cut -d':' -f 1)
+  echo $SUBSTRING_2
+}
+
+function push_to_quay {
+  ID=$(docker run -d $2)
```

Comment:
We should cleanup (e.g. `docker stop -f $ID`) this container after it starts.

Could we possibly achieve this without `docker run`, maybe something like `docker tag`? (I'm no docker expert nor have I put too much thought into this yet.)